src/catalog/catalog.cpp

```diff
@@ -65,12 +65,14 @@ Catalog::Catalog() : pool_(new type::EphemeralPool()) {
   storage_manager->AddDatabaseToStorageManager(peloton);
 
   // Create catalog tables
-  DatabaseCatalog::GetInstance(peloton, pool_.get(), txn);
-  BootstrapSystemCatalogs(peloton, txn);
+  DatabaseCatalog::GetInstance(txn, peloton, pool_.get());
+  BootstrapSystemCatalogs(txn, peloton);
 
   // Insert peloton database into pg_database
-  DatabaseCatalog::GetInstance()->InsertDatabase(
-      CATALOG_DATABASE_OID, CATALOG_DATABASE_NAME, pool_.get(), txn);
+  DatabaseCatalog::GetInstance(nullptr, nullptr, nullptr)->InsertDatabase(txn,
```

Comment:
Why pass a null txn pointer to `DatabaseCatalog::GetInstance()` when you actually have the txn pointer?